### ISSUE TITLE
IAM: Ability to add a clientId to an existing OpenIdConnectProvider

### ISSUE THREAD

**Issue by shaundclements** [aws/aws-cdk#32421]
### Describe the feature

When using the OpenIdConnectProvider it would be great to be able to add ClientIds to an existing provisioned one.
The only supported methods are: 
applyRemovalPolicy(policy)
toString()
static fromOpenIdConnectProviderArn(scope, id, openIdConnectProviderArn)

ref: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.OpenIdConnectProvider.html

### Use Case

When you use the OpenIdConnectProvider.fromOpenIdConnectProviderArn to reference an existing one, it would be really useful to be able to add additional clientIds to support more than one client using the same OIDCProvider.
We are limited in that you can only create one OIDC Providers with the same location/url per account so we end up sharing the same OIDC Provider.
Each environment would then want to add their own ClientId to the created OIDCProvider

### Proposed Solution

`const oidcProvider = aws_iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(
      this,
      'StsWindowsOidcProvider',
      props.oidcProviderArn,
    );
 oidcProvider.addClientId("GUID");

`

### Other Information

It would be good to have the clientIds as a SET so that when a clientId is added (and already there), its not duplicated in a list or breaks the deployment because already there.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.133.0

### Environment details (OS name and version, etc.)

macOS Sonoma. Version: 14.5

**Comment by pahud** [comment]
```
oidcProvider.addClientId("GUID");
```

I am not sure if adding an `addClientId()` for the interface is the way to go but I agree fromOpenIdConnectProviderArn() should allow to specify more options.

**Comment by shaundclements** [comment]
addClientId was my attempt to be consistent with the creation property (clientIds) when first creating an OpenIdConnectProvider, ref: 
<img width="901" alt="image" src="https://github.com/user-attachments/assets/0c6298f9-5df2-479f-8bbe-5381f4f889c5">

**Comment by nmussy** [comment]
> I am not sure if adding an addClientId() for the interface is the way to go

Agreed, a setter would allow to add, remove or update the list of client IDs with just a single method.

It also looks like the custom resource already handles `ClientIDList` updates, so this should be a fairly simple change if you want to give it a shot @shaundclements:

https://github.com/aws/aws-cdk/blob/d1bd1454aad6b4c7e4221bf7960f5f5b6cc3ce6d/packages/@aws-cdk/custom-resource-handlers/lib/aws-iam/oidc-handler/index.ts#L60-L79

**Comment by shaundclements** [comment]
if we went with a setter method, we would also need a read property to get the existing list of ClientIds (so that we can add one more to it)

**Comment by nmussy** [comment]
What I had in mind would overwrite the existing values, either constructed or imported:

```ts
const provider = new OpenIdConnectProvider(stack, 'MyProvider', {
  url: 'https://my-url',
});
provider.clientIds = ['client1'];

// ...
const importedProvider = OpenIdConnectProvider.fromOpenIdConnectProviderArn(stack, 'provider', provider.openIdConnectProviderArn);
importedProvider.clientIds = ['client1', 'client2'];
```

This would allow both adding to the list of clients but also to remove previous values

**Comment by shaundclements** [comment]
This would unfortunately not work for separate environment pipelines which share the same OIDCProvider and just add their clientId to the existing provisioned one. They wont necesarily have knowledge of the other clientIds, so I was after an 'add' method rather than replace.

**Comment by nmussy** [comment]
Sending a list of client IDs to add to the custom resource would clash with its existing `ClientIDList` attribute, I think it would get a bit confusing DX wise.

For your specific use-case, I would say a custom resource of your own is probably the way to go, see the [CDK docs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CustomResource.html) and the existing [OidcProvider](https://github.com/aws/aws-cdk/blob/d1bd1454aad6b4c7e4221bf7960f5f5b6cc3ce6d/packages/@aws-cdk/custom-resource-handlers/lib/aws-iam/oidc-handler/index.ts) handler

**Comment by shaundclements** [comment]
I think extending method support for OpenIdConnectProvider.fromOpenIdConnectProviderArn is more preferrable to workaround suggestion
